examples/wordpress/backend.jsonnet

```diff
@@ -0,0 +1,229 @@
+          },
+          containers_+: {
+            default: kube.Container("mariadb") {
+              image: "bitnami/mariadb",
+              ports_+: {
+                mysql: { containerPort: 3306 },
+              },
+              env_+: {
+                "MARIADB_REPLICATION_MODE": "master",
+                "MARIADB_REPLICATION_USER": { secretKeyRef: { name: "mariadb", key: "replication_user" } },
+                "MARIADB_REPLICATION_PASSWORD": { secretKeyRef: { name: "mariadb", key: "replication_password" } },
+                "MARIADB_ROOT_USER": { secretKeyRef: { name: "mariadb", key: "root_user" } },
+                "MARIADB_ROOT_PASSWORD": { secretKeyRef: { name: "mariadb", key: "root_password" } },
+                "MARIADB_USER": { secretKeyRef: { name: "mariadb", key: "database_user" } },
+                "MARIADB_DATABASE": { secretKeyRef: { name: "mariadb", key: "database_name" } },
+                "MARIADB_PASSWORD": { secretKeyRef: { name: "mariadb", key: "database_password"
+              }}},
+              livenessProbe: {
+                initialDelaySeconds: 40,
+                exec: {
+                  command: [
+                    "sh",
+                    "-c",
+                    "exec mysqladmin status -u$MARIADB_ROOT_USER -p$MARIADB_ROOT_PASSWORD",
+              ]}},
+              readinessProbe: {
+                initialDelaySeconds: 30,
+                exec: {
+                  command: [
+                    "sh",
+                    "-c",
+                    "exec mysqladmin status -u$MARIADB_ROOT_USER -p$MARIADB_ROOT_PASSWORD",
+              ]}},
+              volumeMounts_+: {
+                "mariadb-data": {
+                  "mountPath": "/bitnami/mariadb",
+            }}},
+            metrics: kube.Container("metrics") {
+              image: "prom/mysqld-exporter:v0.10.0",
+              command: [
+                "sh",
+                "-c",
+                "DATA_SOURCE_NAME=\"root:$MARIADB_ROOT_PASSWORD@(localhost:3306)/\" /bin/mysqld_exporter",
+              ],
+              ports_+: {
+                metrics: { containerPort: 9104 },
+              },
+              env_+: {
+                "MARIADB_ROOT_PASSWORD": { secretKeyRef: { name: "mariadb", key: "root_password"
+              }}},
+              livenessProbe: {
+                initialDelaySeconds: 15,
+                timeoutSeconds: 1,
+                httpGet: {
+                  path: "/metrics",
+                  port: 9104,
+              }},
+              readinessProbe: {
+                initialDelaySeconds: 5,
+                timeoutSeconds: 5,
+                httpGet: {
+                  path: "/metrics",
+                  port: 9104,
+      }}}}}},
+      volumeClaimTemplates_+: {
+        "mariadb-data": {
+          storage: "10Gi",
+          metadata+: {
+            labels+: master_labels,
+  }}}}},
+
+  backend_slave_statefulset: kube.StatefulSet("mariadb-slave") {
```

Comment:
I think we should build both mariadb master+slave off some sort of common base template.  I think it would be good for maintenance generally, but even more so here since this is meant to serve as an example of what jsonnet can do.